firebase-firestore/src/main/java/com/google/firebase/firestore/util/ListenerRegistrationImpl.java

```diff
@@ -33,6 +35,115 @@
   /** The event listener for the query that raises events asynchronously. */
   private final ExecutorEventListener<ViewSnapshot> asyncEventListener;
 
+  public static class StopListenerSupportFragment extends android.support.v4.app.Fragment {
```

Comment:
You could reduce duplication here by making something like

```java
class CallbackList {
  public void run() {
    for (Runnable callback : callbacks) {
      if (callback != null) {
        callback.run();
      }
    }
  }
 
  void add(Runnable callback) {
    callbacks.add(callback);
  }

  private final List<Runnable> callbacks = new ArrayList<>();
}

public static class StopListenerSupportFragment  ... {
  @Override
  public void onStop() {
    super.onStop();
    callbacks.run();
  }

  CallbackList callbacks;
}
```

And then just have the code below call `fragment.callbacks.add()`.